File: python_code/dot_prod.R

dot_prod_dist <- function(A, B){
  # function for computing euclidean distances using dot-product
  # Arguements : A and B are vectors of the same length 
  # Outputs: The Euclidean distance between A and B
  
  
  # checking if the vectors are in the same dimension
  if(length(A) != length(B)) stop('Invalid input(s)')
  
  u <- np.asarray(A - B)
  return(sqrt(sum(u*u)))
}

File: python_code/dot_prod.py

def dot_prod(u, v):
    assert(len(u) == len(v), "invalid input(s)")
    return sum(np.array(u)*np.array(v))


def dot_prod_dist(u, v):
    assert(len(u) == len(v), "invalid input(s)")
    A = np.array(u) - np.array(v)
    return sqrt(dot_prod(A, A))
        